pkg/odo/cli/catalog/list/services.go

```diff
@@ -55,6 +55,9 @@ func (o *ServiceOptions) Complete(name string, cmd *cobra.Command, args []string
 		} else if strings.Contains(err.Error(), "cannot list resource \"clusterserviceclasses\" in API group \"servicecatalog.k8s.io\" at the cluster scope") {
 			// this error is thrown when Service Catalog is not enabled on OpenShift
 			err = nil
+		} else if strings.Contains(err.Error(), "cannot list clusterserviceclasses.servicecatalog.k8s.io at the cluster scope") {
```

Comment:
Couldn't we just ignore all errors?
Are there any errors when we would like odo to fail?

Reply:
Don't specifically remember which error we care about here. But yeah, that would make sense as well.

Reply:
I think we care about https://github.com/openshift/odo/blob/e8030b19a3378c0a904aacb6355a6896fee03fed/pkg/catalog/catalog.go#L283-L286

Reply:
My worry is that matching error messages like this is fragile. The message content can change when with different k8s versions.
You can already see this happening. Both error messages that you are ignoring are describing the same issue. It is only formated differently.

Reply:
I agree, and I had the same concern at the time of writing this. But I didn't see another way. If not for https://github.com/openshift/odo/pull/4420#discussion_r574423237, I would have ignored errors altogether.

Reply:
Can we use `kerrors.IsNotFound()` here?

Reply:
> Can we use kerrors.IsNotFound() here?

I think that this is a different situation than "NotFound", but it is worth the errors package as there might be other checking functions that would fit our situation.

@dharmit You don't need to be blocked on this. If you can't find anything better let's go with this. I'm adding lgtm and hold. So you can remove hold yourself if would don't find anything. 


/hold
/lgtm

Reply:
I have removed the error check while fetching Service Catalog based services https://github.com/openshift/odo/pull/4420/commits/fd1e93382efd3ae7fed0fadb079725bdce9d9d2d. IMO, it's not worth spending time on because in all this time that I have worked on `odo service` side of things, I have not hit that particular error.

/hold cancel

Reply:
Just one last thing 😇 Can you add a comment that explains why we are ignoring that error? 
I'm afraid that for the future ourselves, it might not be obvious why in this case we don't care about the error